rostime/src/time.cpp

```diff
@@ -167,6 +167,29 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_monotonictime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+#if HAS_CLOCK_GETTIME
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    // what to do if clock_gettime is not available???
+    struct timeval timeofday;
+    gettimeofday(&timeofday,NULL);
+    sec  = timeofday.tv_sec;
+    nsec = timeofday.tv_usec * 1000;
+#endif
+#else
+    ros_walltime(sec, nsec);
+#endif
```

Comment:
This would be a great place to use `std::chrono::steady_clock` or `std::chrono::high_resolution_clock` when it returns true for `is_steady`, since they're portable.

However, I'm not sure that would be acceptable within `roscpp_core` since it's a C++11 only feature.